### PR TITLE
Fix API hash related crash in `EditorSettings`

### LINKED ISSUE #80062: Assert in `ClassDB::set_current_api` in latest master when launched w/ --verbose
### Godot version

master [75f9c97deaa8378477441e07ea1782ecab6d2eca]

### System information

Godot v4.2.dev (262d1eaa6) - Ubuntu 20.04.6 LTS (Focal Fossa) - X11 - Vulkan (Forward+) - dedicated AMD Radeon RX 6800 XT - AMD Ryzen 9 5900X 12-Core Processor (24 Threads)

### Issue description

Running a dev build on both Linux and Windows reaches the following crash:

```
TextServer: Primary interface set to: "ICU / HarfBuzz / Graphite (Built-in)".
CORE API HASH: 2795494918
EDITOR API HASH: 1481585570
ERROR: FATAL: DEV_ASSERT failed  "!api_hashes_cache.has(p_api)" is false.
   at: set_current_api (core/object/class_db.cpp:59)
```

Call stack (Linux):

```
1 ClassDB::set_current_api                class_db.cpp        59   0x9c77bf6 
2 EditorSettings::ensure_class_registered editor_settings.cpp 883  0x639a96a 
3 EditorNode::EditorNode                  editor_node.cpp     6825 0x61badd9 
4 Main::start                             main.cpp            3085 0x4a6e353 
5 main                                    godot_linuxbsd.cpp  72   0x49c9d0d 
```

Call stack (Windows):

```
>	godot.windows.editor.dev.x86_64.exe!ClassDB::set_current_api(ClassDB::APIType p_api) Line 58	C++
 	godot.windows.editor.dev.x86_64.exe!EditorSettings::ensure_class_registered() Line 886	C++
 	godot.windows.editor.dev.x86_64.exe!EditorNode::EditorNode() Line 6826	C++
 	godot.windows.editor.dev.x86_64.exe!Main::start() Line 3085	C++
 	godot.windows.editor.dev.x86_64.exe!widechar_main(int argc, wchar_t * * argv) Line 179	C++
 	godot.windows.editor.dev.x86_64.exe!_main() Line 204	C++
 	godot.windows.editor.dev.x86_64.exe!main(int argc, char * * argv) Line 218	C++
 	godot.windows.editor.dev.x86_64.exe!WinMain(HINSTANCE__ * hInstance, HINSTANCE__ * hPrevInstance, char * lpCmdLine, int nCmdShow) Line 232	C++
 	[Inline Frame] godot.windows.editor.dev.x86_64.exe!invoke_main() Line 102	C++
```

### Steps to reproduce

Compile with (possibly not all of those flags are needed):

`scons platform=linuxbsd optimize=none debug_symbols=True tests=False dev_mode=True dev_build=True use_llvm=yes use_lld=yes  opengl3=no openxr=no`

Open a project on the editor with:

Working folder: The project's path
Arguments: `-e --verbose`

**The --verbose argument is important!**

### Other info (for bisect)

It must've broken somewhere between 75f9c97deaa8378477441e07ea1782ecab6d2eca and 6588a4a29af1621086feac0117d5d4d37af957fd since it was working fine before that.

**Update:** It's possible I'm wrong about the bisecting info, because the problem started when I added `--verbose` to the cmd line.

### Workaround

Just commenting out the assert runs fine.

### Minimal reproduction project

Any.

### PR DESCRIPTION
**Pull request by AThousandShips**
Crash due to manipulating API after getting the hash, which happens with `--verbose`

Instead of registering `EditorSettings` as abstract, and then registering as complete when creating an instance, we now register it as complete from the start.

This behavior is equivalent to how it worked prior to #78615, where the built-in documentation doesn't show the default values of `EditorSettings`, the abstract state of `EditorSettings` is and has always been false, and it is and has been possible to instance it from scripts (required for the loading of the settings)

I don't think the loss of the built-in documentation for these settings is a major loss, `EditorSettings` itself remains in the documentation, and the results of `--doctool` remains the same

In summary, I think this is a good balanced solution to the issue, without having to mess with the specific registration location of `EditorSettings`. Unless there is some reason for it being abstract that I have missed (other than the documentation generation)

* Fixes: #80062

### CODE REVIEW

**Review by RandomShaper — APPROVED** [review]
Seems to fix the issue rationally. However, I'd still like more eyes looking into this.

**Comment by YuriSizov** [comment]
> I don't think the loss of the built-in documentation for these settings is a major loss, `EditorSettings` itself remains in the documentation, and the results of `--doctool` remains the same

I don't understand from this description what we are losing here. Just the default values? We don't have them documented anyway...

> and it is and has been possible to instance it from scripts (required for the loading of the settings)

`EditorSettings` should not be instanced from scripts. It should be accessed through the `EditorInterface` instance.

**Comment by AThousandShips** [comment]
> I don't understand from this description what we are losing here. 

They weren't documented before, in current master have values documented only in built-in

> EditorSettings should not be instanced from scripts. 

Agreed, but this is possible in the current code, don't think we can block it without also losing the ability to load it, think this needs a larger rework

**Comment by YuriSizov** [comment]
> They weren't documented before, in current master have values documented only in built-in, so not anything lost IMO

Oh, you mean that in the built-in help we get the default values added despite them not being present in the XML docs? Well, that could be a problem if we want to show default values in on-hover tooltips. But we don't right now, so I agree that this is fine.

> Agreed, but this is possible in the current master, don't think we can block it without also losing the ability to load it, think this needs a larger rework

We don't need to block it. But we shouldn't consider this as a valid use case either. Though currently the docs for `EditorSettings` are missing a mention that you should access them through the interface, I think. Worth adding, as we do it in some other classes.

**Comment by AThousandShips** [comment]
>  Though currently the docs for EditorSettings are missing a mention that you should access them through the interface, I think. 

They already do fortunately:
> Note: This class shouldn't be instantiated directly. Instead, access the singleton using [EditorInterface.get_editor_settings](https://docs.godotengine.org/en/stable/classes/class_editorinterface.html#class-editorinterface-method-get-editor-settings).

Thank you for checking these things, important questions

As far as I know creating a new `EditorSettings` instance on your own doesn't have any sideeffects (though it does seem to affect some index counting that might break things, can be worth investigating, but this PR has no effect on that either way)

**Review by akien-mga — APPROVED** [review]
Makes sense. It's somewhat cleaner than the previous code registering EditorSettings twice.

**Comment by akien-mga** [comment]
Thanks!

**Comment by AThousandShips** [comment]
Thank you!
